### PR TITLE
Topology clean ups - Part 2

### PR DESCRIPTION
**Pull request by ranj063**
Rename host gateway pipelines
Make copier node_type mandatory

### CODE REVIEW

File: tools/topology/topology2/include/pipelines/cavs/host-gateway-capture.conf [hunk 1]
```diff
@@ -1,12 +1,12 @@
 #
-# CAVS passthrough capture pipeline
+# Host gateway capture pipeline
 #
-# A simple passthrough pipeline. All attributes defined herein are namespaced
-# by alsatplg to "Object.Pipeline.passthrough-capture.N.attribute_name".
+# A simple host gateway pipeline for playback. All attributes defined herein are namespaced
```
Comment (plbossart) [inline tools/topology/topology2/include/pipelines/cavs/host-gateway-capture.conf:4]: capture here?

File: tools/topology/topology2/include/pipelines/cavs/dai-copier-be.conf [hunk 2]
```diff
@@ -72,6 +72,7 @@ Class.Pipeline."dai-copier-be" {
 		copier."2" {
 			type	buffer
 			copier_type	"module"
+			node_type	$BUFFER_COPIER
```
Comment (plbossart) [inline tools/topology/topology2/include/pipelines/cavs/dai-copier-be.conf:75]: why do we need a dai-copier-be in the first place? or is this again a case of a pipeline name not describing well what it actually does?

File: tools/topology/topology2/include/pipelines/cavs/dai-copier-eqiir-module-copier-capture.conf [hunk 3]
```diff
@@ -86,6 +86,7 @@ Class.Pipeline."dai-copier-eqiir-module-copier-capture" {
 		copier."2" {
 			type buffer
 			copier_type	"module"
+			node_type	$BUFFER_COPIER
```
Comment (plbossart) [inline tools/topology/topology2/include/pipelines/cavs/dai-copier-eqiir-module-copier-capture.conf:89]: that pipeline should be broken-up in two: io-gateway and processing pipelines.

File: tools/topology/topology2/include/pipelines/cavs/gain-copier-capture.conf [hunk 4]
```diff
@@ -92,6 +92,7 @@ Class.Pipeline."gain-copier-capture" {
 		copier."2" {
 			type buffer
 			copier_type	"module"
+			node_type	$BUFFER_COPIER
```
Comment (plbossart) [inline tools/topology/topology2/include/pipelines/cavs/gain-copier-capture.conf:95]: awful names again, sometimes there's a module-copier but not always.

File: tools/topology/topology2/include/pipelines/cavs/dai-copier-be.conf (hunk 2 above)
Comment (plbossart) [inline tools/topology/topology2/include/pipelines/cavs/dai-copier-be.conf:75]: can't we have this attribute set as immutable in the base class? Or have a different copier-module class?

**Review by plbossart — COMMENTED** [review]
looks mostly good

**Comment by sys-pt1s** [comment]
Can one of the admins verify this patch?

File: tools/topology/topology2/include/pipelines/cavs/dai-copier-eqiir-module-copier-capture.conf (hunk 3 above)
Comment (ranj063) [reply tools/topology/topology2/include/pipelines/cavs/dai-copier-eqiir-module-copier-capture.conf]: yes i want to do that for both the host and dai sides. coming soon

File: tools/topology/topology2/include/pipelines/cavs/dai-copier-be.conf (hunk 2 above)
Comment (ranj063) [reply tools/topology/topology2/include/pipelines/cavs/dai-copier-be.conf]: i wanted to take the sort-cut but yes makes better sense to have a new class. fixed now

**Review by lgirdwood — APPROVED** [review]

**Review by plbossart — COMMENTED** [review]
LGTM, just one doubt on node_type, see below.

File: tools/topology/topology2/include/components/copier.conf [hunk 5]
```diff
@@ -169,6 +168,7 @@ Class.Widget."copier" {
 			"num_output_pins"
 			"num_input_audio_formats"
 			"num_output_audio_formats"
+			"node_type"
```
[the remark this answers is not in the document]
Comment (ranj063) [reply tools/topology/topology2/include/components/copier.conf:171]: it is already set everywhere @plbossart. Just adding this constraint for the future

**Review by plbossart — APPROVED** [review]
LGTM

**Comment by ranj063** [comment]
@wszypelt Can you please check the internal CI?

**Comment by wszypelt** [comment]
@ranj063 There was a problem with the internal service, we managed to fix it, I will rerun PR

**Comment by kv2019i** [comment]
PM failures in https://sof-ci.01.org/sofpr/PR7416/build5662/devicetest/index.html and https://sof-ci.01.org/sofpr/PR7416/build5663/devicetest/index.html . Do not seem related to this PR, so proceeding with merge. Rest of CI is good,

**Pull request merged**